lib/elixir/lib/gen_server.ex

```diff
@@ -479,8 +485,8 @@ defmodule GenServer do
   If `reason` is not `:normal`, `:shutdown`, nor `{:shutdown, term}` an error is
   logged.
   """
-  @callback terminate(reason, state :: term) ::
-    term when reason: :normal | :shutdown | {:shutdown, term} | term
+  @callback terminate(reason, state :: term) :: term
+            when reason: :normal | :shutdown | {:shutdown, term}
```

Comment:
Don't understand why the formatter moves `term` to the previous line... visually this is weird. If we have to expand these lines, I would expect the following to pass the formatter:

```elixir
@callback terminate(reason, state :: term) ::
            term when reason: :normal
            | :shutdown
            | {:shutdown, term}
```

But it get's converted to:

```elixir
@callback terminate(reason, state :: term) :: term
            when reason: :normal
                         | :shutdown
                         | {:shutdown, term}
```